src/main/java/org/influxdb/dto/Point.java

```diff
@@ -553,13 +564,37 @@ private void formatedTime(final StringBuilder sb, final TimeUnit precision) {
     if (this.time == null) {
       return;
     }
-    if (precision == null) {
-      sb.append(" ").append(TimeUnit.NANOSECONDS.convert(this.time, this.precision));
-      return;
+    TimeUnit converterPrecision = precision;
+
+    if (converterPrecision == null) {
+      converterPrecision = TimeUnit.NANOSECONDS;
+    }
+    if (this.time instanceof BigInteger) {
```

Comment:
I am a bit afraid of the performance impact, this is in the hot path and instanceof is not known for beeing super performant. Can we prove somehow, how big the performance impact might be ?

Reply:
I created a benchmark to measure the impact. It can be found here: https://github.com/J0s3f/influx-point-benchmark
To run it on you own machine, simply download [the jar](https://github.com/J0s3f/influx-point-benchmark/releases/download/1.0/benchmarks.jar) and run it via `java -jar benchmarks.jar`

Here are the results on my machine: https://gist.github.com/J0s3f/6e6feb1cef2644bdb060576e99b1e91f

I'd say there is no meaningful impact of this change.

Reply:
Its about 4% slower ! Given that users complain about performance already, i dont want to take extra without double check if this is not possible in another way

Reply:
If you take the numbers from my benchmark, benchmarkNewPoint is the changed code so it is 4% faster (but with bigger error margin)

Reply:
Cant image the extra code speeds up the processing ?

Reply:
Benchmarks are just not accurate to that level on any PC that's running a full blown desktop OS.
I can try to increase the Forks in JMH but I don't have access to a bare metal machine I can use to run that for a few hours currently :(

And I changed one if in the existing code, this might have impact on the JIT compiler optimizations.

Reply:
I made another benchmark using the settings `java -jar /benchmarks.jar -bm thrpt -f 5 -i 8 -si true -t max -wi 5` on a Quad-Core Intel Xeon VM that wasn't running anything else. (of course, I don't know what the host system was doing).

Here are the results: https://gist.github.com/J0s3f/379f9d2a1cfd6b77a5931b9422e80e42

Reply:
OK, granted.

Can you please add some Unit tests to have these new logic covered as well: https://codecov.io/gh/influxdata/influxdb-java/compare/45f6cf44396acbf38161ab27b2c53ffbd2894fbf...fcc5a8fdc889d73af4a8495bad921db51c1dd33e/diff#D1-x558

Reply:
Done